src/thing_doors.c

```diff
@@ -101,11 +122,48 @@ char get_door_orientation(MapSlabCoord slb_x, MapSlabCoord slb_y)
     }
 }
 
-struct Thing *create_door(struct Coord3d *pos, unsigned short a1, unsigned char a2, unsigned short a3, unsigned char a4)
+struct Thing *create_door(struct Coord3d *pos, ThingModel tngmodel, unsigned char orient, PlayerNumber plyr_idx, TbBool is_locked)
 {
-  return _DK_create_door(pos, a1, a2, a3, a4);
+    struct Thing *doortng;
+    struct DoorStats* dostat = &door_stats[tngmodel][orient];
+
+    if ( i_can_allocate_free_thing_structure(FTAF_FreeEffectIfNoSlots) )
+    {
+        doortng = allocate_free_thing_structure(FTAF_FreeEffectIfNoSlots);
+
+        doortng->class_id = TCls_Door;
+        doortng->model = tngmodel;
+        doortng->mappos.x.val = pos->x.val;
+        doortng->mappos.y.val = pos->y.val;
+        doortng->mappos.z.val = 384;
+        doortng->next_on_mapblk = 0;
+        doortng->parent_idx = doortng->index;
+        doortng->owner = plyr_idx;
+        doortng->field_4F |= TF4F_Unknown01;
+        doortng->door.orientation = orient;
+        doortng->active_state = DorSt_Closed;
+        doortng->creation_turn = game.play_gameturn;
+        doortng->health = dostat->health;
+        doortng->door.is_locked = is_locked;
+
+        add_thing_to_its_class_list(doortng);
+        place_thing_in_mapwho(doortng);
+
+        place_animating_slab_type_on_map(dostat->slbkind, 0,  doortng->mappos.x.stl.num, doortng->mappos.y.stl.num, plyr_idx);
+        //update_navigation_triangulation(stl_x-1,  stl_y-1, stl_x+2,stl_y+2);
+        if ( game.neutral_player_num != plyr_idx )
+            ++game.dungeon[plyr_idx].total_doors;
+        return doortng;
+    }
+    else
+    {
+        ERRORLOG("error! Cannot create door because there are too many fucking things allocated.");
```

Comment:
`ERRORDBG(3,"Cannot create door for player %d. There are too many things allocated.",(int)owner);`

There's an error message in `i_can_allocate_free_thing_structure` already.

Reply:
I believe I did the same for the effect generator, I'll fix it in both then